### DISCUSSION TITLE
Is there a way to get the main image of a list of listings

### DISCUSSION THREAD

**Discussion by VasilVelikov** [etsy/open-api#861] [Q&A] [answered]
Hi everyone,

I am looking for a way to get the main image of a list of listings with a single API call. Basically, I have a list of 100 active listing_ids and I want to see their thumbnail images without making 100 separate API calls. In v2 we were able to request a MainImage association in the findAllListingsActive endpoint. However, in v3 it seems like this is not possible.
Can you please advise if there is some solution or a workaround?
Thank you!

**Comment by Gareth064** [comment, marked as answer]
What endpoint are you calling?

If you are calling any of the following ShopListing endpoints then you can include the images


  *   GetListingsByShop
  *   GetListing
  *   GetListingsbyListingIds

If you are getting your listing data from anything else then you will need to use the GetListingImages endpoint, passing the listing_id. And do this for all 100 listins.

If you are getting Receipts, their transactions have a listing_id and a listing_image_id which you can use with GetListingImage.


Kind Regards

Gareth Doherty



It’s nice to be important,

But it’s more important to be nice.



________________________________
From: VasilVelikov ***@***.***>
Sent: 09 February 2023 22:08
To: etsy/open-api ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [etsy/open-api] Is there a way to get the main image of a list of listings (Discussion #861)


Hi everyone,

I am looking for a way to get the main image of a list of listings with a single API call. Basically, I have a list of 100 active listing_ids and I want to see their thumbnail images without making 100 separate API calls. In v2 we were able to request a MainImage association in the findAllListingsActive endpoint. However, in v3 it seems like this is not possible.
Can you please advise if there is some solution or a workaround?
Thank you!

**Reply by VasilVelikov** [reply]
Thank you for the details. I've missed when you've added includes to GetListingsbyListingIds. This solves my problem.
Thank you!

**Comment by Gareth064** [comment]
` I have a list of 100 active listing_ids and I want to see their thumbnail images without making 100 separate API calls`
if you already have the listing_ids then you can use `GetListingsbyListingIds` passing in a comma separate list of the Id's and includes images. You would then get all 100 listings (with all their data) and their image data in 1 call. You just need to choose the first image from the image array which I assume is the primary image.

**Comment by careybaird** [comment]
Also waiting for this
